R/methods-MChromatograms.R

```diff
@@ -354,3 +277,309 @@ setMethod("filterColumnsKeepTop", "MChromatograms",
+#'
+#' @param ylim optional `numeric(2)` defining the y-axis limits.
+#'
+#' @param ... optional arguments to be passed to the plotting functions (see
+#'     help on the base R `plot` function.
+#'
+#' @return silently returns a `list` (length equal to `ncol(object)` of
+#'     `numeric` (length equal to `nrow(object)`) with the y position of
+#'     each EIC.
+#'
+#' @md
+#'
+#' @author Johannes Rainer
+#'
+#' @name plotChromatogramsOverlay
+#'
+#' @examples
+#'
+#' ## Load preprocessed data and extract EICs for some features.
+#' library(xcms)
+#' data(xdata)
+#' ## Update the path to the files for the local system
+#' dirname(xdata) <- c(rep(system.file("cdf", "KO", package = "faahKO"), 4),
+#'                     rep(system.file("cdf", "WT", package = "faahKO"), 4))
+#' ## Subset to the first 3 files.
+#' xdata <- filterFile(xdata, 1:3, keepFeatures = TRUE)
+#'
+#' ## Define features for which to extract EICs
+#' fts <- c("FT097", "FT163", "FT165")
+#' chrs <- featureChromatograms(xdata, features = fts)
+#'
+#' plotChromatogramsOverlay(chrs)
+#'
+#' ## plot the overlay of EICs in the first sample
+#' plotChromatogramsOverlay(chrs[, 1])
+#'
+#' ## Define a different color for each feature (row in chrs). By default, also
+#' ## all chromatographic peaks of a feature is labeled in the same color.
+#' plotChromatogramsOverlay(chrs[, 1],
+#'     col = c("#ff000040", "#00ff0040", "#0000ff40"))
+#'
+#' ## Alternatively, we can define a color for each individual chromatographic
+#' ## peak and provide this with the `peakBg` and `peakCol` parameters.
+#' chromPeaks(chrs[, 1])
+#'
+#' ## Use a color for each of the two identified peaks in that sample
+#' plotChromatogramsOverlay(chrs[, 1],
+#'     col = c("#ff000040", "#00ff0040", "#0000ff40"),
+#'     peakBg = c("#ffff0020", "#00ffff20"))
+#'
+#' ## Plotting the data in all samples.
+#' plotChromatogramsOverlay(chrs,
+#'     col = c("#ff000040", "#00ff0040", "#0000ff40"))
+#'
+#' ## Creating a "stacked" EIC plot: the EICs are placed along the y-axis
+#' ## relative to their m/z value. With `stacked = 1` the y-axis is split in
+#' ## half, the lower half being used for the stacking of the EICs, the upper
+#' ## half being used for the *original* intensity axis.
+#' res <- plotChromatogramsOverlay(chrs[, 1], stacked = 1,
+#'     col = c("#ff000040", "#00ff0040", "#0000ff40"))
+#' ## add horizontal lines for the m/z values of each EIC
+#' abline(h = res[[1]], col = "grey", lty = 2)
+#'
+#' ## Note that this type of visualization is different than the conventional
+#' ## plot function for chromatographic data, which will draw the EICs for
+#' ## multiple samples into the same plot
+#' plot(chrs)
+#'
+#' ## Converting the object to a MChromatograms without detected peaks
+#' chrs <- as(chrs, "MChromatograms")
+#'
+#' plotChromatogramsOverlay(chrs,
+#'     col = c("#ff000040", "#00ff0040", "#0000ff40"))
+setMethod("plotChromatogramsOverlay", "MChromatograms",
```

Comment:
I was wondering if optional log-scaling and/or normalisation would be interesting, to better reveal small peak shapes (e.g. 2nd or 3rd isotope ...)

Reply:
Good point. I will add a parameter `transform = identify` that would allow the user to e.g. specify `transform = log` to log transform the data.